### PR TITLE
test: add more glados tests for sync

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **Refactor**
  * Refined sync conflict classification so state comparisons are more decisive and consistent during synchronization.

* **Tests**
  * Added extensive property-based tests covering vector clocks, outbox unpacking, pagination, sent-event registry, inbound queue/worker, and pending decryption to improve reliability and catch edge cases.

### CODE REVIEW

File: test/features/sync/matrix/sent_event_registry_test.dart [hunk 1]
```diff
@@ -15,6 +18,165 @@ class _MutableClock extends Clock {
...
+
+extension _AnySentRegistryScenario on glados.Any {
+  glados.Generator<_GeneratedRegistryOperationKind> get registryOperationKind =>
+      glados.AnyUtils(this).choose(_GeneratedRegistryOperationKind.values);
+
+  glados.Generator<_GeneratedRegistryOperation> get registryOperation =>
+      glados.CombinableAny(this).combine3(
+        registryOperationKind,
+        glados.IntAnys(this).intInRange(0, 5),
+        glados.IntAnys(this).intInRange(0, 7),
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/sync/matrix/sent_event_registry_test.dart:154]: fixed
